make 64bit compliant

ken3 opened the pull request:
If pcf2bdf is built and run on LP64 environment, int32 value can be wrong.
Why int32 is typedef-ed as long? It would be better to typedef int32 as int.

--- jiskan24.bdf.org    2017-09-01 19:16:59.241457019 +0900
+++ jiskan24.bdf        2017-09-01 19:08:13.041455456 +0900
@@ -21,7 +21,7 @@
 CHARSET_REGISTRY "JISX0208.1983"
 CHARSET_ENCODING "0"
 WEIGHT 10
-X_HEIGHT 4294967295
+X_HEIGHT -1
 QUAD_WIDTH 24
 DEFAULT_CHAR 8481
 FONT_DESCENT 2

ken3 committed:
make 64bit compliant

ken3 committed:
make 64bit compliant

ganaware commented:
Thank you  for your request.
Your request contains non-essential modifications about braces after "if".
Please include essential modifications only.

(Now I think it is bad habit to omit braces after "if", but my 15-years-younger self liked to omit them.)

ken3 committed:
revert if-else style.

ken3 commented:
OK, I reverted if-parentheses.
I also fixed version comparison issue. See following:

pcf2bdf.cc: In function 'int main(int, char**)':
pcf2bdf.cc:622:45: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (version != make_int32(1, 'f', 'c', 'p'))
                                             ^

ken3 committed:
Merge branch '64bit-compliant' of https://github.com/ken3/pcf2bdf into 64bit-compliant

ken3 commented:
rebased.

ganaware commented:
Thanks!
